vm/vm.go

```diff
@@ -279,6 +279,32 @@ func (vm *Engine) validateStateTransition(virtualTx *wire.MsgTx) error {
 		return newErrKind(ErrInvalidTransferWitness)
 	}
 
+	// Initialite an array holding all the assets that appear in this state
+	// transition outputs.
+	assetOutputs := make([]*asset.Asset, 0, len(vm.splitAssets)+1)
+
+	// Add the root asset to the list of outputs.
+	// TODO(george): was the root asset placed 1st on the signer's side? is
+	// this enforced by bips?
+	assetOutputs = append(assetOutputs, vm.newAsset)
+
+	// Add the split assets to the list of outputs.
+	// TODO(geoge): this function states that there's no splits, does that
+	// mean that following loop is a noop?
+	for _, splitAsset := range vm.splitAssets {
+		assetOutputs = append(assetOutputs, &splitAsset.Asset)
+	}
+
+	// Construct the virtual transaction that this input signed,
+	// based on the sighash flag of the signature that we acquired
+	// above.
+	virtualTx, _, err := tapscript.VirtualTx(
+		vm.newAsset, vm.prevAssets, assetOutputs,
```

Comment:
This should take the script version as input. We can't break all the old versions (but again, I don't think this level of changes are needed: output sighashes are all or nothing).